api/controllers/articles-controllers.js

```diff
@@ -11,12 +12,38 @@ exports.getArticlesById = (req,res,next) =>{
     })
   }
 
-exports.getAllArticles = (req,res,next) => {
-    selectAllArticles()
-    .then((articles) => {
-        res.status(200).send({articles})
-    })
-    .catch(next)
+  exports.getAllArticles = (req,res,next) => {
+    if(Object.keys(req.query).length !== 0){
+        const {topic} = req.query;
+        if (topic){
+            Promise.all([selectAllArticles(topic),checkTopicExists(topic)])
+            .then((resolvedArticles) => {
+                const articles = resolvedArticles[0]
+                if(articles.length === 0) {
+                    return Promise.reject({status: 404, msg: 'Article Not Found'})
+                }
+                res.status(200).send({articles})
+            })
+            .catch((err)=> {
+                next(err)
+            })
+        } else {
+            return Promise.reject({status: 400, msg: 'Bad Request'})
+            .catch((err)=> {
+                 next(err)
+            })
```

Comment:
Your checkTopicExists model will return an error if the topic doesn't exist, so you shouldn't need this Promise.reject here